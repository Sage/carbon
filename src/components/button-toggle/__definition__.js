import ButtonToggle from './button-toggle.component';
import Definition from './../../../demo/utils/definition';
import OptionsHelper from 'utils/helpers/options-helper';

let definition = new Definition('button-toggle', ButtonToggle, {
  description: `Selects one option from a small number of highly visible options.`,
  designerNotes: `
* Useful if the user has a choice between a small number of options, that youâ€™d like to be highly visible (e.g. choosing between three or fewer subscription packages, with short names).
* To make the meaning of an option clearer, you can add an icon to it. Just name one of the Carbon icons.
  `,
  relatedComponentsNotes: `
* Performing a single action? [Try Button](/components/button).
* Range of buttons where one is more important? [Try Split Button](/components/split-button).
* Range of buttons of the same importance? [Try Multi Action Button](/components/multi-action-button).
* Choosing one option from a longer list? [Try Radio Button](/components/radio-button).
* Choosing one option from a very long list? [Try Dropdown](/components/dropdown).
* Choosing more than one option? [Try Checkbox](/components/checkbox).
* Need a visual? [Try Icons](/style/icons).
 `,
  type: 'action',
  numberOfExamples: 3,
  propOptions: {
    buttonIcon: OptionsHelper.icons,
    buttonIconSize: OptionsHelper.sizesBinary,
    size: OptionsHelper.sizesBinary
  },
  propValues: {
    name: "option",
    children: "Option",
    value: "option-1"
  },
  propTypes: {
    buttonIcon: "String",
    buttonIconSize: "String",
    disabled: "Boolean",
    grouped: "Boolean",
    name: "String",
    size: "String",
    value: "String",
    children: "Node"
  },
  requiredProps: ['children'],
  propRequires: {
    buttonIconSize: "buttonIcon"
  },
  propDescriptions: {
    buttonIcon: "Define an icon to use for the button",
    buttonIconSize: "Define what size the icon should be. Possible values include: " + OptionsHelper.sizesBinary.join(", "),
    disabled: "Disable all user interaction.",
    grouped: "Removes the spacing between buttons.",
    name: "The label that will be used for the selected button. Use the same name across multiple buttons for them to operate as radio buttons.",
    size: "Define what size the button should be. This will default to large if not set. Possible values include: " + OptionsHelper.sizesBinary.join(", "),
    value: "The value associated with the selected button.",
    children: "The children to render for the button."
  },
  hiddenProps: ["name", "value"]
});

export default definition;
