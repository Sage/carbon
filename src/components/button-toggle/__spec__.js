import React from 'react';
import TestUtils from 'react/lib/ReactTestUtils';
import { shallow } from 'enzyme';
import ButtonToggle from './button-toggle';

describe('ButtonToggle', () => {
  let instance, wrapper;

  beforeEach(() => {
    instance = TestUtils.renderIntoDocument(<ButtonToggle><span>Plus</span></ButtonToggle>);
    wrapper = shallow(
      <ButtonToggle data-role='contacts' data-element='button-toggle'><span>Minus</span></ButtonToggle>
    );
  });

  describe('mainClasses', () => {
    it('returns the classes for the component', () => {
      expect(instance.mainClasses).toEqual('carbon-button-toggle common-input');
    });
  });

  describe('input classes', () => {
    it('returns the classes for the input', () => {
      expect(instance.inputClasses).toEqual('carbon-button-toggle__input hidden common-input__input');
    });
  });

  describe('buttonIcon', () => {
    describe('with no buttonIcon', () => {
      it('returns nothing', () => {
        expect(instance.buttonIcon).toBe(null);
      });
    });

    describe('with an buttonIcon', () => {
      beforeEach(() => {
        instance = TestUtils.renderIntoDocument(
          <ButtonToggle buttonIcon='settings'><span>Plus</span></ButtonToggle>
        );
      });

      it('returns the buttonIcon', () => {
        expect(instance.buttonIcon.props.className).toEqual('carbon-button-toggle__button-icon');
      });
    });

    describe('with a large buttonIcon', () => {
      beforeEach(() => {
        instance = TestUtils.renderIntoDocument(
          <ButtonToggle buttonIcon='settings' buttonIconSize='large'><span>Plus</span></ButtonToggle>
        );
      });

      it('returns a large buttonIcon', () => {
        expect(instance.buttonIcon.props.className).toEqual(
          'carbon-button-toggle__button-icon carbon-button-toggle__button-icon--large'
        );
      });
    });
  });

  describe('additionalInputContent', () => {
    it('returns the label', () => {
      expect(instance.additionalInputContent.props.className).toEqual('carbon-button-toggle__label');
    });

    describe('if it is disabled', () => {
      it('returns a disabled label', () => {
        instance = TestUtils.renderIntoDocument(
          <ButtonToggle disabled={ true }><span>Plus</span></ButtonToggle>
        );

        expect(instance.additionalInputContent.props.className).toEqual(
          'carbon-button-toggle__label carbon-button-toggle__label--disabled'
        );
      });
    });
  });

  describe('label id', () => {
    it('assigns the guid as the label id', () => {
      expect(instance.inputProps.id).toEqual(instance._guid);
    });

    it('assigns a custom id if one is given', () => {
      instance = TestUtils.renderIntoDocument(<ButtonToggle id='foo'><span>Plus</span></ButtonToggle>);
      expect(instance.inputProps.id).toEqual('foo');
    });
  });

  describe("tags", () => {
    describe("on component", () => {
      let wrapper = shallow(<ButtonToggle data-element='bar' data-role='baz'>Test</ButtonToggle>);

      it('include correct component, element and role data tags', () => {
        window.RootTagTest.run(wrapper, 'button-toggle', 'bar', 'baz');
      });
    });

    describe("on internal elements", () => {
      let wrapper = shallow(<ButtonToggle>Test</ButtonToggle>);

      window.ElementsTagTest.run(wrapper, [
        'input',
        'label'
      ]);
    });
  });
});
