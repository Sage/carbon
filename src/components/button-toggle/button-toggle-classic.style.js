import { css } from 'styled-components';
import { THEMES } from '../../style/themes';

const StyledButtonToggleClassicLabel = ({ theme }) => theme.name === THEMES.classic && css`
  height: auto;
  min-width: auto;
  padding: 15px 25px;
  font-weight: 900;
  border: 1px solid #ccd6db;

  input:checked ~ & {
    color: ${theme.colors.white};
    background-color: #1573e6;
  }

  input:focus ~ & {
    outline: 0;
  }  
  
  &:hover {
    border-color: #1e499f;
    color: ${theme.colors.white};
    background-color: #1e499f;
  }

  ${({ size }) => size === 'small' && css`
    height: auto;
    padding: 5px 8px;
    font-weight: 700;
    font-size: 12px;
  `};

  ${({ size, buttonIcon, buttonIconSize }) => buttonIcon && size === 'large' && buttonIconSize === 'large' && css`
    height: auto;
    padding-top: 15px;
    padding-bottom: 15px;

    .carbon-button-toggle__button-icon {
      margin-right: 3px;
    }
  `};
`;

const StyledButtonToggleClassicIcon = ({ theme }) => theme.name === THEMES.classic && css`
  display: inline;

  ${({ buttonIconSize }) => buttonIconSize === 'large' && css`
    display: block;

    .carbon-icon {
      margin-bottom: 0;
    }
  `};

  ${({ buttonIcon, buttonIconSize }) => buttonIcon && buttonIconSize === 'large' && css`
      margin-right: 3px;
  `};
`;

export {
  StyledButtonToggleClassicLabel,
  StyledButtonToggleClassicIcon
};
