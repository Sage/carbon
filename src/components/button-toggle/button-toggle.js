import React from 'react';
import classNames from 'classnames';
import Icon from 'components/icon';
import css from 'utils/css';
import Input from 'utils/decorators/input';
import { validProps } from 'utils/ether';
import { tagComponent } from '../../utils/helpers/tags';


const ButtonToggle = Input(
class ButtonToggle extends React.Component {
  static propTypes = {
    /**
     * Which buttonIcon the button should render.
     *
     * @property buttonIcon
     * @type {String}
     */
    buttonIcon: React.PropTypes.string,

    /**
     * Sets the size of the buttonIcon (eg. large)
     *
     * @property buttonIconSize
     * @type {String}
     */
    buttonIconSize: React.PropTypes.string,

    /**
     * Defines the name for the input
     *
     * @property name
     * @type {String}
     */
    name: React.PropTypes.string,

    /**
     * The value for the given button.
     *
     * @property value
     * @type {String}
     */
    value: React.PropTypes.string,

    /**
     * A required prop. This is what the button will display.
     *
     * @property children
     * @type {Multiple}
     */
    children: React.PropTypes.node.isRequired
  }

  static safeProps = ['name']

  /**
   * Main Class getter
   *
   * @method mainClasses
   * @return {void}
   */
  get mainClasses() {
    return 'carbon-button-toggle';
  }

  /**
   * Uses the inputClasses method provided by the decorator to add additional classes.
   *
   * @method inputClasses
   * @return {String} input className
   */
  get inputClasses() {
    return classNames('carbon-button-toggle__input', css.hidden);
  }

  /**
   * Returns the markup for the buttonIcon.
   *
   * @method buttonIcon
   * @return {Object} JSX
   */
  get buttonIcon() {
    if (!this.props.buttonIcon) { return null; }

    let classes = classNames('carbon-button-toggle__button-icon', {
      ['carbon-button-toggle__button-icon--large']: this.props.buttonIconSize === 'large'
    });

    return (
      <div className={ classes } data-element='icon'>
        <Icon type={ this.props.buttonIcon } />
      </div>
    );
  }

  /**
   * A getter that combines props passed down from the input decorator with
   * textbox specific props.
   *
   * @method inputProps
   * @return {Object} props for the input
   */
  get inputProps() {
    let { ...props } = validProps(this);
    delete props.children;
    props.className = this.inputClasses;
    props.type = 'radio';
    if (!props.id) {
      props.id = this._guid;
    }
    return props;
  }

  /**
   * Returns additional content to sit inline with the input.
   *
   * @method additionalInputContent
   * @return {Object} JSX
   */
  get additionalInputContent() {
    let classes = classNames('carbon-button-toggle__label', {
      ['carbon-button-toggle__label--disabled']: this.props.disabled
    });

    return (
      <label htmlFor={ this.inputProps.id } className={ classes } data-element='label'>
        { this.buttonIcon }
        { this.props.children }
      </label>
    );
  }

  /**
   * @method render
   * @return {Object} JSX
   */
  render() {
    return (
      <div className={ this.mainClasses } { ...tagComponent('button-toggle', this.props) }>
        { this.inputHTML }
      </div>
    );
  }
}
);

export default ButtonToggle;
