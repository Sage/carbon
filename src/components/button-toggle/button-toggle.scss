@import "./../../style-config/colors";
@import "./../../style-config/buttons";

.carbon-button-toggle {
  display: inline-block;

  + .carbon-button-toggle {
    margin-left: 10px;
  }
}

.carbon-button-toggle__label {
  background-color: $white;
  border-radius: $carbon-button__border-radius;
  border: 1px solid $grey-dark-blue-20;
  color: $grey-dark-blue-60;
  cursor: pointer;
  display: inline-block;
  font-weight: bold;
  padding: 15px 25px;
  text-align: center;

  &:hover {
    background-color: $pod-footer;
  }
}

.carbon-button-toggle__label--disabled,
.carbon-button-toggle__label--disabled:hover {
  background-color: $carbon-button__background-color--disabled !important;
  color: $carbon-button__color--disabled !important;
  cursor: initial;
}

.carbon-button-toggle__input:checked ~ .carbon-button-toggle__label {
  background-color: $blue-sky;
  border-color: $blue-sky-20;
  color: $white;
}

.carbon-button-toggle__button-icon {
  display: inline-block;
  top: -2px;
  margin-right: 3px;
  position: relative;
}

.carbon-button-toggle__button-icon--large {
  display: block;

  .carbon-icon:before {
    font-size: 60px;
    line-height: 60px;
  }

  .carbon-icon__svg {
    height: 64px;
    width: 64px;
  }
}
