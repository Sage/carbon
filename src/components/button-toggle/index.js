export { default } from './button-toggle.component';
