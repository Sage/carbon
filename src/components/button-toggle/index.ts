export { default as ButtonToggle } from "./button-toggle.component";
export type { ButtonToggleProps } from "./button-toggle.component";
export { default as ButtonToggleGroup } from "./button-toggle-group";
export type { ButtonToggleGroupProps } from "./button-toggle-group";
