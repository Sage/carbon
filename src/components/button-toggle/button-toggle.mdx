import { Meta, ArgTypes, Canvas } from "@storybook/blocks";

import * as ButtonToggleStories from "./button-toggle.stories";
import * as ButtonToggleGroupStories from "./button-toggle-group/button-toggle-group.stories";

<Meta title="Button Toggle" of={ButtonToggleStories} />

# Button Toggle

<a
  target="_blank"
  href="https://zeroheight.com/2ccf2b601/p/996a28-button-toggle/b/7384f9"
  style={{ color: "#007E45", fontWeight: "bold", textDecoration: "underline" }}
  rel="noreferrer"
>
  Product Design System component
</a>

Press one of the buttons to make selection. This component should be used when user has to make a choice between a small number of options.

## Contents

- [Quick Start](#quick-start)
- [Examples](#examples)
- [Props](#props)

## Quick Start

```javascript
import {
  ButtonToggle,
  ButtonToggleGroup,
} from "carbon-react/lib/components/button-toggle";
```

## Examples

### Default

<Canvas of={ButtonToggleStories.Default} />

### With inputHint

<Canvas of={ButtonToggleStories.InputHint} />

### Controlled

<Canvas of={ButtonToggleStories.Controlled} />

### With aria-label

If you do not want a visible label to appear above the group, you should use the aria-label prop to give it a non-visible
accessible name to be used by screen readers. Failure to do this will result in accessibility violations.

<Canvas of={ButtonToggleStories.AriaLabel} />

### With fullWidth

When the `fullWidth` prop is `true`, the buttons will expand in size to take up the full width of the container.

<Canvas of={ButtonToggleStories.FullWidth} />

### Allowing deselection

By default, in a `ButtonToggleGroup` there is no way to deselect the currently-selected toggle button except by selecting another one.
The `allowDeselect` prop changes this behaviour, making it possible for there to be no selection.

If you use this, you should include text in the `inputHint` prop which makes clear that deselection is allowed, as in the example below.

<Canvas of={ButtonToggleStories.AllowDeselection} />

### Small icon

<Canvas of={ButtonToggleStories.DefaultSmallIcon} />

### Large icon

<Canvas of={ButtonToggleStories.DefaultLargeIcon} />

### Icon only

While we do not advocate the use of icon-only buttons for accessibility reasons, the `aria-label` or `aria-labelledby` props can be used with `buttonIcon` so assistive techniques are aware of the action that will be taken when the component pressed.

<Canvas of={ButtonToggleStories.IconOnly} />

### Small

<Canvas of={ButtonToggleStories.Small} />

### Small with small icon

<Canvas of={ButtonToggleStories.SmallSmallIcon} />

### Small with large icon

<Canvas of={ButtonToggleStories.SmallLargeIcon} />

### Large

<Canvas of={ButtonToggleStories.Large} />

### Large with small icon

<Canvas of={ButtonToggleStories.LargeSmallIcon} />

### Large with large icon

<Canvas of={ButtonToggleStories.LargeLargeIcon} />

### Disabled Button

<Canvas of={ButtonToggleStories.DisabledButton} />

### Disabled Group

<Canvas of={ButtonToggleStories.DisabledGroup} />

### Wrapped Group

Buttons will wrap onto the next line in smaller screens.

<Canvas of={ButtonToggleStories.WrappedButtons} />

## Props

### Button Toggle

<ArgTypes of={ButtonToggleStories} />

### Button Toggle Group

<ArgTypes of={ButtonToggleGroupStories} />
