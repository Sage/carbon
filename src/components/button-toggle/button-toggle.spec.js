import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import TestRenderer from 'react-test-renderer';
import guid from '../../utils/helpers/guid';
import { classicTheme, baseTheme } from '../../style/themes';
import ButtonToggle from './button-toggle.component';
import ButtonToggleInput from './button-toggle-input.component';
import { assertStyleMatch, carbonThemesJestTable } from '../../__spec_helper__/test-utils';
import { StyledButtonToggleIcon, StyledButtonToggleLabel, StyledButtonToggleInput } from './button-toggle.style';
import { InputGroupContext } from '../../__internal__/input-behaviour';

jest.mock('../../utils/helpers/guid');
guid.mockImplementation(() => 'guid-12345');

describe('ButtonToggle', () => {
  describe('functionality', () => {
    it('pass onChange props to input', () => {
      const onChangeMock = jest.fn();
      const wrapper = renderWithTheme({
        theme: baseTheme,
        onChange: onChangeMock
      });

      wrapper.find(ButtonToggleInput).prop('onChange')();
      expect(onChangeMock.mock.calls.length).toBe(1);
    });

    it('pass onBlur props to input', () => {
      const onBlurMock = jest.fn();
      const wrapper = renderWithTheme({
        theme: baseTheme,
        onBlur: onBlurMock
      });

      wrapper.find(ButtonToggleInput).prop('onBlur')();
      expect(onBlurMock.mock.calls.length).toBe(1);
    });

    it('pass onFocus props to input', () => {
      const onFocusMock = jest.fn();
      const wrapper = renderWithTheme({
        theme: baseTheme,
        onFocus: onFocusMock
      });

      wrapper.find(ButtonToggleInput).simulate('focus');
      expect(onFocusMock.mock.calls.length).toBe(1);
    });
  });

  describe('when a label is clicked', () => {
    const onClickMock = jest.fn();
    let wrapper;
    let domWrapper;

    beforeEach(() => {
      domWrapper = document.createElement('div');
      document.body.appendChild(domWrapper);
      wrapper = mount(<ButtonToggle onClick={ onClickMock }>Button</ButtonToggle>,
        { attachTo: domWrapper });
    });

    afterEach(() => {
      wrapper.detach();
      document.body.removeChild(domWrapper);
    });

    it('then the input should be focused', () => {
      wrapper.find(StyledButtonToggleLabel).simulate('click');
      expect(wrapper.update().find(ButtonToggleInput).getDOMNode()).toEqual(document.activeElement);
    });
  });

  describe('HiddenCheckableInput', () => {
    let propOnBlur;
    let groupContextOnBlur;

    let groupContextOnFocus;

    let groupContextOnMouseEnter;

    let groupContextOnMouseLeave;

    let wrapper;

    beforeEach(() => {
      propOnBlur = jest.fn();
      groupContextOnBlur = jest.fn();

      groupContextOnFocus = jest.fn();

      groupContextOnMouseEnter = jest.fn();

      groupContextOnMouseLeave = jest.fn();

      wrapper = renderWithContext(
        {
          onBlur: propOnBlur
        },
        {
          onBlur: groupContextOnBlur,
          onFocus: groupContextOnFocus,
          onMouseEnter: groupContextOnMouseEnter,
          onMouseLeave: groupContextOnMouseLeave
        },
      );
    });


    it('triggers onFocus callbacks passed from props and context', () => {
      wrapper.find(StyledButtonToggleInput).props().onFocus();
      expect(groupContextOnFocus).toHaveBeenCalled();
    });

    it('triggers onBlur callbacks passed from props and context', () => {
      wrapper.find(StyledButtonToggleInput).props().onBlur();
      expect(propOnBlur).toHaveBeenCalled();
      expect(groupContextOnBlur).toHaveBeenCalled();
    });

    it('triggers onMouseEnter callback passed from context', () => {
      wrapper.find(StyledButtonToggleLabel).props().onMouseEnter();
      expect(groupContextOnMouseEnter).toHaveBeenCalled();
    });

    it('triggers onMouseLeave callback passed from context', () => {
      wrapper.find(StyledButtonToggleLabel).props().onMouseLeave();
      expect(groupContextOnMouseLeave).toHaveBeenCalled();
    });

    it('does nothing if onBlur callbacks are not provided', () => {
      wrapper = renderWithContext();
      const inputProps = wrapper.find(StyledButtonToggleInput).props();

      inputProps.onBlur();
    });

    it('does nothing if onFocus callbacks are not provided', () => {
      wrapper = renderWithContext();
      const inputProps = wrapper.find(StyledButtonToggleInput).props();

      inputProps.onFocus();
    });
  });

  describe('Classic theme', () => {
    it('renders correctly with default settings', () => {
      const wrapper = renderWithTheme({ theme: classicTheme }, TestRenderer.create);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly with large buttonIcon and large size', () => {
      const wrapper = renderWithTheme({
        theme: classicTheme,
        buttonIcon: 'add',
        buttonIconSize: 'large',
        size: 'large'
      });
      assertStyleMatch({
        height: 'auto',
        paddingTop: '15px',
        paddingBottom: '15px'
      }, wrapper.find('label'));
    });
    it('renders correctly with small size', () => {
      const wrapper = renderWithTheme({
        theme: classicTheme,
        size: 'small'
      });
      assertStyleMatch({
        height: 'auto',
        padding: '5px 8px',
        fontWeight: '700',
        fontSize: '12px'
      }, wrapper.find('label'));
    });
  });
  describe.each(carbonThemesJestTable)('when the %s theme is set', (themeName, theme) => {
    it('renders correct styles', () => {
      const wrapper = renderWithTheme({
        theme
      }, TestRenderer.create);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correct styles for a large icon', () => {
      const wrapper = renderWithTheme({
        theme,
        buttonIcon: 'add',
        buttonIconSize: 'large'
      });
      assertStyleMatch({
        minWidth: '104px',
        height: '102px',
        padding: '0 16px'
      }, wrapper.find('label'));
    });
  });
  describe('General styling', () => {
    it('renders correctly when disabled', () => {
      const wrapper = renderWithTheme({
        theme: classicTheme,
        disabled: true
      });
      assertStyleMatch({
        backgroundColor: '#E6EBED',
        borderColor: '#E6EBED',
        color: 'rgba(0,0,0,.2)'
      }, wrapper.find('label'), { modifier: '&' });
    });
    it('renders correctly with small icon', () => {
      const wrapper = renderWithTheme({
        theme: classicTheme,
        buttonIcon: 'add',
        buttonIconSize: 'small'
      });
      assertStyleMatch({
        marginRight: '8px'
      }, wrapper.find(StyledButtonToggleIcon));
    });
    it('renders correctly when grouped', () => {
      const props = {
        grouped: true,
        children: 'Text'
      };
      const wrapper = TestRenderer.create(
        <ThemeProvider theme={ classicTheme }>
          <div>
            <ButtonToggle { ...props } />
            <ButtonToggle { ...props } />
          </div>
        </ThemeProvider>
      );
      // Uses snapshot as jest/enzyme doesnt support :first-of-type
      expect(wrapper).toMatchSnapshot();
    });
  });
});

function renderWithTheme(props = {}, renderer = mount) {
  const { theme, ...componentProps } = props;

  return renderer(
    <ThemeProvider theme={ theme }>
      <ButtonToggle { ...componentProps }>Button</ButtonToggle>
    </ThemeProvider>
  );
}

function renderWithContext(props = {}, inputGroupContextValue = {}) {
  return mount(
    <InputGroupContext.Provider value={ inputGroupContextValue }>
      <ButtonToggle { ...props }>Button</ButtonToggle>
    </InputGroupContext.Provider>
  );
}
