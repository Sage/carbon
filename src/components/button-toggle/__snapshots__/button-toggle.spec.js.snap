// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonToggle Classic theme renders correctly with default settings 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c2 {
  display: inline-block;
  position: relative;
  height: 38px;
  padding: 0 24px;
  font-weight: 600;
  cursor: pointer;
  border: 1px solid #668592;
  font-size: 14px;
  background-color: #FFFFFF;
  height: auto;
  min-width: auto;
  padding: 15px 25px;
  font-weight: 900;
  border: 1px solid #ccd6db;
}

input:checked ~ .c2 {
  color: #FFFFFF;
  background-color: #00402E;
}

input:checked ~ .c2 .c4 {
  color: #FFFFFF;
}

input:focus ~ .c2 {
  outline: 3px solid #FFB500;
  z-index: 100;
}

.c2:hover {
  background-color: #E6F5E6;
  border-color: #00402E;
}

input:checked ~ .c2 {
  color: #FFFFFF;
  background-color: #1573e6;
}

input:checked ~ .c2 .c4 {
  color: #FFFFFF;
}

input:focus ~ .c2 {
  outline: 0;
}

.c2:hover {
  border-color: #1e499f;
  color: #FFFFFF;
  background-color: #1e499f;
}

.c2:hover .c4 {
  color: #FFFFFF;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c0:not(:first-of-type) {
  margin-left: 10px;
}

.c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

<div
  className="c0"
  data-component="button-toggle"
  onClick={[Function]}
>
  <input
    className="c1"
    id="guid-12345"
    onBlur={[Function]}
    onFocus={[Function]}
    type="radio"
  />
  <label
    className="c2"
    htmlFor="guid-12345"
    size="large"
  >
    <div
      className="c3"
    >
      Button
    </div>
  </label>
</div>
`;

exports[`ButtonToggle General styling renders correctly when grouped 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c2 {
  display: inline-block;
  position: relative;
  height: 38px;
  padding: 0 24px;
  font-weight: 600;
  cursor: pointer;
  border: 1px solid #668592;
  font-size: 14px;
  background-color: #FFFFFF;
  height: auto;
  min-width: auto;
  padding: 15px 25px;
  font-weight: 900;
  border: 1px solid #ccd6db;
}

input:checked ~ .c2 {
  color: #FFFFFF;
  background-color: #00402E;
}

input:checked ~ .c2 .c8 {
  color: #FFFFFF;
}

input:focus ~ .c2 {
  outline: 3px solid #FFB500;
  z-index: 100;
}

.c2:hover {
  background-color: #E6F5E6;
  border-color: #00402E;
}

input:checked ~ .c2 {
  color: #FFFFFF;
  background-color: #1573e6;
}

input:checked ~ .c2 .c8 {
  color: #FFFFFF;
}

input:focus ~ .c2 {
  outline: 0;
}

.c2:hover {
  border-color: #1e499f;
  color: #FFFFFF;
  background-color: #1e499f;
}

.c2:hover .c8 {
  color: #FFFFFF;
}

.c5 .c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 .c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 .c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c0:not(:first-of-type) {
  margin-left: 10px;
}

.c0:not(:first-of-type) {
  margin-left: -1px;
}

.c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

<div>
  <div
    className="c0"
    data-component="button-toggle"
    onClick={[Function]}
  >
    <input
      className="c1"
      id="guid-12345"
      onBlur={[Function]}
      onFocus={[Function]}
      type="radio"
    />
    <label
      className="c2"
      htmlFor="guid-12345"
      size="large"
    >
      <div
        className="c3 c4"
      >
        Text
      </div>
    </label>
  </div>
  <div
    className="c0"
    data-component="button-toggle"
    onClick={[Function]}
  >
    <input
      className="c1"
      id="guid-12345"
      onBlur={[Function]}
      onFocus={[Function]}
      type="radio"
    />
    <label
      className="c2"
      htmlFor="guid-12345"
      size="large"
    >
      <div
        className="c3 c4"
      >
        Text
      </div>
    </label>
  </div>
</div>
`;

exports[`ButtonToggle when the aegean theme is set renders correct styles 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c5 .c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 .c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: inline-block;
  position: relative;
  height: 38px;
  padding: 0 24px;
  font-weight: 600;
  cursor: pointer;
  border: 1px solid #668592;
  font-size: 14px;
  background-color: #FFFFFF;
}

input:checked ~ .c2 {
  color: #FFFFFF;
  background-color: #004472;
}

input:checked ~ .c2 .c7 {
  color: #FFFFFF;
}

input:focus ~ .c2 {
  outline: 3px solid #FFB500;
  z-index: 100;
}

.c2:hover {
  background-color: #E6F1FA;
  border-color: #004472;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c0:not(:first-of-type) {
  margin-left: 10px;
}

.c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

<div
  className="c0"
  data-component="button-toggle"
  onClick={[Function]}
>
  <input
    className="c1"
    id="guid-12345"
    onBlur={[Function]}
    onFocus={[Function]}
    type="radio"
  />
  <label
    className="c2"
    htmlFor="guid-12345"
    size="large"
  >
    <div
      className="c3 c4"
    >
      Button
    </div>
  </label>
</div>
`;

exports[`ButtonToggle when the mint theme is set renders correct styles 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c5 .c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: inline-block;
  position: relative;
  height: 38px;
  padding: 0 24px;
  font-weight: 600;
  cursor: pointer;
  border: 1px solid #668592;
  font-size: 14px;
  background-color: #FFFFFF;
}

input:checked ~ .c2 {
  color: #FFFFFF;
  background-color: #00402E;
}

input:checked ~ .c2 .c6 {
  color: #FFFFFF;
}

input:focus ~ .c2 {
  outline: 3px solid #FFB500;
  z-index: 100;
}

.c2:hover {
  background-color: #E6F6F1;
  border-color: #00402E;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c0:not(:first-of-type) {
  margin-left: 10px;
}

.c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

<div
  className="c0"
  data-component="button-toggle"
  onClick={[Function]}
>
  <input
    className="c1"
    id="guid-12345"
    onBlur={[Function]}
    onFocus={[Function]}
    type="radio"
  />
  <label
    className="c2"
    htmlFor="guid-12345"
    size="large"
  >
    <div
      className="c3 c4"
    >
      Button
    </div>
  </label>
</div>
`;
